worlds/kh1/Client.py

```diff
@@ -256,14 +268,15 @@ async def game_watcher(ctx: KH1Context):
                     if st != "nil":
                         if timegm(time.strptime(st, '%Y%m%d%H%M%S')) > ctx.last_death_link and int(time.time()) % int(timegm(time.strptime(st, '%Y%m%d%H%M%S'))) < 10:
                             await ctx.send_death(death_text = "Sora was defeated!")
-                if file.find("insynthshop") > -1:
-                    if not ctx.hinted_synth_location_ids:
+                if file.find("hint") > -1:
+                    hint_location_id = int(file.split("hint", -1)[1])
+                    if hint_location_id not in ctx.hinted_location_ids:
                         await ctx.send_msgs([{
                             "cmd": "LocationScouts",
-                            "locations": [2656401,2656402,2656403,2656404,2656405,2656406],
+                            "locations": [hint_location_id],
                             "create_as_hint": 2
                         }])
-                        ctx.hinted_synth_location_ids = True
+                        ctx.hinted_location_ids.append(hint_location_id)
         ctx.locations_checked = sending
         message = [{"cmd": 'LocationChecks', "locations": sending}]
```

Comment:
I'd recommend using self.check_locations(sending) or testing len(sending) here to avoid spamming the server every game update